### DISCUSSION TITLE
Mapping Automation - Edit/Change Order or Number

### DISCUSSION THREAD

**Discussion by bluetardis** [apohl79/audiogridder#1396] [Ideas]
I am trying to map Channel Strip (eg BrainWorx 4000E) to some hardware/knobs.

The automation shows up on the controller if I have chosen Assign All but the order/workflow isn't good. (_This isn't an audio gridder issue, it's just how the plugin was programmed_.).

Its very hard to get lots (50 parameters or so) mapped in the correct order as the window closes each time and if you make a mistake you pretty much have to start over.

**I would like a way to edit or assign the automation mapping order.**  

**Ideas I have are**
Option 1: Save all the Plugin Automation in a file (when you save it as an AutoGridder Preset) thats basically the Name of the Parameter and its assigned order.  This could then be edited.  I know this somewhat exists in the .preset file but its really tricky to edit without corruption

Option2: Don't close the window when assigning automation manually.  This may we could add/remove parameters easily

Option3: Assign All but have an option to raise/lower the priority. (This way everything is mapped and we can raise/lower)

Not sure what's easiest to implement from your side but anything would be appreciated.

**Comment by oferfuxake** [comment]
I'd like to second this, but extend the suggestion to all of the menus in this section of the plugin. I remove all audio channel assignments aside from L/R on all of my plugins for performance and for the sake of my DAW's happiness & stability. Having to re-enter the menu to do this is indeed annoying. If the menus would just stay open until clicked out of, it would really help streamline this kind of thing. I also customize the automation assignments like @bluetardis does. My workaround has been to set my plugins up the way I want to and just save them as a preset, so it's not like I'm doing this on every session or every instance, but if it's easy to implement it'd be great.

With that said, AG is absolutely _amazing_ and has completely changed the way I work. It's the kind of solution that is normally so expensive that only high-end users could afford it. @apohl79 is in my pantheon of heroes now.

**Reply by oferfuxake** [reply]
Oh, just to add - reducing the active audio channels to only the ones I'm using has made a HUGE improvement in performance. As in, I'm running stable on 2 blocks now where it used to take 12 or more. Plugin meters actually move at the same time I hear the sound now, and when I move a control I don't have to wait to hear its effect. I highly recommend only using the audio channels you need! This might be a nice addition to some of the setup / troubleshooting information. I haven't seen it referenced anywhere yet, though I may have just overlooked it.

**Reply by bluetardis** [reply]
I stumbled across this solution in my workflow (not using the side chains) and glad to see others found it as well. 

I will raise another suggestion. Probably that there is a toggle so that by default only the mains in/out is used as that works for most of the time and it’s easy enough to add a side chain when needed. 

Maybe it’s just be but I tend to turn off sandboxing too. I have a stack of channel strip plugins and with sandbox off they can “talk” to each other for global changes.

**Reply by oferfuxake** [reply]
I absolutely agree that it'd be a great feature to be able to toggle for L/R channels only (or even better, to be the default setting.) Not many people are working on surround mixes, and if someone is they can always turn the other channels on. The vast majority of us are just doing stereo.

In my setup, I'm not just eliminating the sidechains - I've had to manually deselect 28 of my 30 audio in / out channels on each plugin. (See the screen pictures.) I don't know if the high channel count is due to my DAW, my audio interface, or something else - does this long list appear in your AG menu?

![20241009_210051](https://github.com/user-attachments/assets/6e2757d4-f889-4e33-a765-e5d3f19348e9)
![20241009_210138](https://github.com/user-attachments/assets/96427bca-d45e-4a8b-ae4b-9cfa0ade41ce)

Also - I'm really interested in the benefits you have found from turning off sandboxing. I've got mine on the default "chain sandboxing" setting but now I'm going to experiment with that. Could you describe your use of the channel strips talking to each other and making global changes? I'm interested in trying out the capability but I'm not clear on how you're using it.

**Reply by bluetardis** [reply]
Turning off Sandboxing examples... I use a lot of Plugin-Alliance Channel Strips.
Most of them model the actual schematic diagrams and each component specification can deviate from the engineering tolerance.  This means that each channel will be subtly different if you want it.  You can also turn this off if you want everything to be exactly the same.

However if you want to use it as old school desk for the entire desk or sections of the mix (eg picking a flavour such as API/SSL/Neve etc) then when this applied across a lot of channels it helps with the overall mix.
When the plugs can see other iterations of themselves, they can manage this for you.  Also it's possible to use a set Global for input levels and other stuff.  This only works when the plugs aren't sand boxed.

Hope thats clear as mud.

**Reply by oferfuxake** [reply]
Those channel strips seem like they'd be really great. Thanks for the information on sandboxing - I never would have experimented with turning it off!
